packages/flutter/lib/src/rendering/stack.dart

```diff
@@ -480,6 +480,11 @@ class RenderStack extends RenderBox
     assert(_resolvedAlignment != null);
     _hasVisualOverflow = false;
     bool hasNonPositionedChildren = false;
+    if (childCount == 0) {
+      size = constraints.biggest;
+      assert(size.isFinite);
+      return;
```

Comment:
did you mean to add this block? it seems to be redundant with the code below, but if you meant it as an optimisation it seems to be in a weird place (since it makes earlier statements redundant in the short-circuited case).

Reply:
oh, i see the comment later about this.

We shouldn't add this kind of thing unless we have benchmarks showing it's actually a benefit.
If we do want to add it we should really do it seriously, which would mean putting it before the hasNonPositionedChildren, at least, and maybe before _resolve (and moving _hasVisualOverflow earlier), assuming _resolve doesn't do anything useful in the case of there being no children.

Also, the assert is redundant with the same assert in box.dart.

Reply:
OK, makes sense.  Want me to revert this?

Reply:
Or, I can follow up with the yak shave and do a performance analysis of Stack.

Reply:
Unless we have numbers showing that we need this, we should probably revert it, and if there's a performance concern, we can then do the yak shave.